proposals/graphql/api/src/main/java/org/eclipse/microprofile/graphql/Argument.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.graphql;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Controls the mapping of a method's parameter to an argument of a GraphQL operation (query/mutation/subscription).
+ * <br><br>
+ * For example, a user might annotate a method's parameter as such:
+ * <pre>
+ * public class CharacterService {
+ *     {@literal @}Query(value = "searchByName",
+ *                 description = "Search characters by name")
+ *     public List{@literal <}Character{@literal >} getByName(
+ *                      {@literal @}Argument(name = "name", defaultValue = "Han Solo", description = "Name to search for") String name) {
+ *         //...
+ *     }
+ * }
+ * </pre>
+ *
+ * Schema generation of this would result in a stanza such as:
+ * <pre>
+ * type Query {
+ *         # Search characters by name
+ *         # name: Name to search for. Default value: Han Solo.
+ *         searchByName(name: String = "Han Solo"): [Character]
+ *     }
+ * </pre>
+ */
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface Argument {
+
+    /**
+     * @return the name to use for the GraphQL argument.
+     */
+    String value() default "";

```

Comment:
By default, in our annotations, in such a case, the Java name is used for the GraphQL name.

Reply:
Ok, I think that would be fine for class name, not sure if you would have access to the argument name ?

Reply:
I don't know for sure, I imagine that if your annotation is on an argument, you should have the details of it, no ? Maybe @kaqqao (Bojan) knows better on that.

Reply:
In my implementation, the name is optional and the Java argument name you get with reflection is used by default. It will log a warning if the names are lost in compilation and suggest compiling with the `-parameters` option.
Should we keep this behavior or make the name mandatory?

Reply:
+1, ok.